libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpH263Reader.java

```diff
@@ -0,0 +1,197 @@
+    if ((header & 0x200) != 0 || (header & 0x1f8) != 0 || (header & 0x7) != 0) {
+      Log.w(TAG, "Packet discarded due to (VRC != 0) or (PLEN != 0) or (PEBIT != 0)");
+      return;
+    }
+    int startCodePayload = data.peekUnsignedByte() & 0xfc;
+    if (pBit == true) {
+      if (startCodePayload < 128) {
+        Log.w(TAG, "Picture start Code (PSC) missing, Dropping packet.");
+        return;
+      } else {
+        // Setting first two bytes of the start code. Refer RFC4629 Section 5.1.
+        data.getData()[currPosition] = 0;
+        data.getData()[currPosition + 1] = 0;
+        data.setPosition(currPosition);
+      }
+    } else {
+      // Check that this packet is in the sequence of the previous packet.
+      int expectedSequenceNumber = RtpPacket.getNextSequenceNumber(previousSequenceNumber);
+      if (sequenceNumber != expectedSequenceNumber) {
+        Log.w(
+            TAG,
+            Util.formatInvariant(
+                "Received RTP packet with unexpected sequence number. Expected: %d; received: %d."
+                    + " Dropping packet.",
+                expectedSequenceNumber, sequenceNumber));
+        return;
+      }
+    }
+
+    if (fragmentedSampleSizeBytes == 0) {
+      getBufferFlagsAndResolutionFromVop(data, isOutputFormatSet);
+      if (!isOutputFormatSet && isKeyFrame == true) {
+        if (width != payloadFormat.format.width || height != payloadFormat.format.height) {
+          trackOutput.format(
+              payloadFormat.format.buildUpon().setWidth(width).setHeight(height).build());
+        }
+        isOutputFormatSet = true;
+      }
+    }
+    int fragmentSize = data.bytesLeft();
+    // Write the video sample.
+    trackOutput.sampleData(data, fragmentSize);
+    fragmentedSampleSizeBytes += fragmentSize;
+
+    if (rtpMarker) {
+      if (firstReceivedTimestamp == C.TIME_UNSET) {
+        firstReceivedTimestamp = timestamp;
+      }
+      long timeUs = toSampleUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp);
+      trackOutput.sampleMetadata(
+          timeUs,
+          isKeyFrame ? C.BUFFER_FLAG_KEY_FRAME : 0,
+          fragmentedSampleSizeBytes,
+          /* offset= */ 0,
+          /* encryptionData= */ null);
+      fragmentedSampleSizeBytes = 0;
+      isKeyFrame = false;
+    }
+    previousSequenceNumber = sequenceNumber;
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
+    firstReceivedTimestamp = nextRtpTimestamp;
+    fragmentedSampleSizeBytes = 0;
+    startTimeOffsetUs = timeUs;
+  }
+
+  // Internal methods.
+  /**
+   * Parses VOP Coding type and resolution.
+   */
+  private void getBufferFlagsAndResolutionFromVop(ParsableByteArray data, boolean gotResolution) {
+    // Search for SHORT_VIDEO_START_MARKER (0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0).
+    int currPosition = data.getPosition();
+    if (data.readUnsignedInt() >> 10 == 0x20) {
```

Comment:
assign `data.readUnsignedInt()` to a variable.

Make sure you `&` and then `>>` and then compare in case there's sign extension.